### PR TITLE
Pubdev 5652 H2O xgboost native xgboost compare

### PR DESCRIPTION
**Pull request by wendycwong**
Added comparison tests for H2OXGBoost and native XGBoost for

binomial classifier
multinomial
regression

with numerical datasets only.

### CODE REVIEW

File: h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_binomial_compare_large.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    print("H2OXGBoost train time is {0}ms.  Native XGBoost train time is {1}s\n.  H2OGBoost scoring time is {2}s."
+          "  Native XGBoost scoring time is {3}s.".format(h2oTrainTimeD, nativeTrainTime,
+                                                                             h2oPredictTimeD, nativeScoreTime))
+    # Result comparison in terms of actual prediction value between the two
+    h2oPredictD['predict'] = h2oPredictD['predict'].asnumeric()
+    h2oPredictLocalD = h2oPredictD.as_data_frame(use_pandas=True, header=True)
+
+    # compare prediction probability and they should agree if they use the same seed
+    for ind in range(h2oPredictD.nrow):
+        assert abs(h2oPredictLocalD['c0.l1'][ind]-nativePred[ind])<1e-6, "H2O prediction prob: {0} and native " \
```
Comment (Pscheidl) [inline h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_binomial_compare_large.py]: Why is ε 0.000001 ? Is this arbitrary ?

File: h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_multinomial_compare_large.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          "  Native XGBoost scoring time is {3}s.".format(h2oTrainTimeD, nativeTrainTime,
+                                                                             h2oPredictTimeD, nativeScoreTime))
+    # Result comparison in terms of actual prediction value between the two
+    h2oPredictD['predict'] = h2oPredictD['predict'].asnumeric()
+    h2oPredictLocalD = h2oPredictD.as_data_frame(use_pandas=True, header=True)
+    nclass = len(nativePred[0])
+    colnames = h2oPredictD.names
+
+    # compare prediction probability and they should agree if they use the same seed
+    for ind in range(h2oPredictD.nrow):
```
Comment (Pscheidl) [inline h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_multinomial_compare_large.py]: Mybe we could have these methods extracted elswhere on a unified place, but this is not an issue now I believe. If nothing else, we'd have central place to change comparison precision at.

File: h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_binomial_compare_large.py (hunk 1 above)
Comment (wendycwong) [reply h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_binomial_compare_large.py]: Good question, this is usually what the other guys have been using.

File: h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_multinomial_compare_large.py (hunk 2 above)
Comment (wendycwong) [reply h2o-py/tests/testdir_algos/xgboost/pyunit_H2OXGBoost_native_XGBoost_numerical_multinomial_compare_large.py]: yes, my thought exactly.  Now that I have some time, I will start doing it.  Getting tired of copying it all over the place.

**Comment by wendycwong** [comment]
tests here are already covered in another PR that has been merged.

**Pull request closed without merging**